lib/CL/devices/ventus/pocl_ventus.cc

```diff
@@ -1101,6 +1101,178 @@ void pocl_ventus_write(void *data,
 }
 
 
+void
+pocl_ventus_read_rect (void *data, void *__restrict__ const host_ptr,
+                       pocl_mem_identifier *src_mem_id, cl_mem src_buf,
+                       const size_t *__restrict__ const buffer_origin,
+                       const size_t *__restrict__ const host_origin,
+                       const size_t *__restrict__ const region,
+                       size_t const buffer_row_pitch,
+                       size_t const buffer_slice_pitch,
+                       size_t const host_row_pitch,
+                       size_t const host_slice_pitch)
+{
+    cl_mem_flags flags = src_buf->flags;
+    void *__restrict__ device_ptr = src_mem_id->mem_ptr;
+
+    if (flags & CL_MEM_USE_HOST_PTR) {
```

Comment:
Does this situation need to be considered individually? When creating the buffer, did we not write from the host to the device buffer when encountering CL_MEM_USE_HOST_PTR?